src/rebuild/README.md

```diff
@@ -1,275 +1,150 @@
-
-As described earlier, when the Raft leader finds that all targets have finished the rebalancing process, it should send a collective RPC as the global completion notification of the rebalance. After receiving the notification, the old targets should exit from the rebalancing mode and reject I/O requests to those migrated objects; the clients who learned of the completion from any server should also exit from the rebalancing mode and only deliver I/O request to new targets. However, given the asynchrony of collective RPC, the DSR protocol should handle some corner cases for this final step.
-
-Let�s examine this scenario: A new target has accepted some writes for an object from Client A, which has just exited from the rebalancing mode. In the meantime, an old target has not heard about the global completion yet, and it has served a read request of the object replica from Client B, who is also in the rebalancing mode. In this case, Client B cannot see the completed write from Client A.
-
-In order to resolve this race condition, the DSR needs to add these steps to the rebalance  protocol:
-
-- If a target has exited from the rebalancing mode, it should piggyback this information on I/O replies to clients.
-- If a client in the rebalancing mode submits an update request to both old and new targets, and learns from their replies that any of those targets have exited from the rebalancing mode, this client should send extra pings to those targets that are still in the rebalancing mode, and request them to exit from the rebalancing mode. After that, this client itself should exit from the rebalancing mode, and complete the update operation.
-
-In this way, the rebalance protocol can guarantee that old targets can always reject I/O requests if they have stale data. The following <a href="#f10.21">figure</a> is an example of this protocol.
-
-<a id="f10.21"></a>
-**Write in rebalancing mode**
-![../../doc/graph/Fig_062.png](../../doc/graph/Fig_062.png "Write in rebalancing mode")
-
-<a id="10.6.3"></a>
-### No Multi-Rebalance
-
-To simplify the design, multi-rebalance is not supported in this design. The user has to serialize pool-map extending operations.
+#Rebuild Detection
+
+When a target failed, it should be detected promptly and
+notify the pool (Raft) leader. then the leader will exclude
+the target from the pool and trigger the rebuild process
+immediately.
+
+##Current status and longterm goal
+
+Currently, since the raft leader can not exclude the target
+automatically, the sysadmin has to manually exclude the target
+from the pool, which then trigger the rebuild.
+
+In future, the leader should be able to detect the target failure
+promptly and then trigger the rebuild automatically by itself,
+without the help of sysadmin.
+
+##multiple pool and targets rebuild
+
+If there are multiple pools being impacted by the failing
+target, these pools can be rebuilt concurrently.
+
+If there are multiple targets in the same pool failes before the
```

Comment:
s/failes/fail, as it is "targets"
and below line s/happenes/happens
BTW, setting ":set spell" in .vimrc can help to figure out some typo